ANTLR/src/Compiler/ContextualAnalysis/TypeCheckerVisitor.java

```diff
@@ -199,6 +194,36 @@ public Integer visitFuncdef(FuncdefContext ctx) {
         return returnType;
     }
 
+    private <T extends ParserRuleContext> ArrayList<Integer> getListOfTypes(T node, Integer size){
+        switch (node.getRuleContext().getRuleIndex()) {
+            case BuffParser.RULE_exprparams:
+                return listCreatedByLoop((ExprparamsNotEmptyContext)node, size);
+            case BuffParser.RULE_funcdef:
+                return listCreatedByLoop((FuncdefContext)node, size);
+        };
+        return null;
+    }
+
+    private ArrayList<Integer> listCreatedByLoop(ExprparamsNotEmptyContext node, Integer size) {
+        ArrayList<Integer> types = new ArrayList<>();
+        for (int i = 0; i < size; i++){
+
+            Integer type = visit((node).expr(i));
```

Comment:
Kunne den her visit statement ikke gives med som funktionsparameter? Så vil de to listCreatedByLoop kunne slås sammen.

Reply:
Tror ikke jeg er helt med på hvad der menes, måske vi lige kan tage den på disc eller mandag